ResidentsSocialCarePlatformApi/V1/UseCase/GetResidentRecordsUseCase.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Linq;
+using ResidentsSocialCarePlatformApi.V1.Boundary.Responses;
+using ResidentsSocialCarePlatformApi.V1.Factories;
+using ResidentsSocialCarePlatformApi.V1.UseCase.Interfaces;
+
+namespace ResidentsSocialCarePlatformApi.V1.UseCase
+{
+    public class GetResidentRecordsUseCase : IGetResidentRecordsUseCase
+    {
+        private readonly IGetVisitInformationByPersonId _getVisitInformationByPersonId;
+        private readonly IGetAllCaseNotesUseCase _getAllCaseNotesUseCase;
+
+        public GetResidentRecordsUseCase(
+            IGetVisitInformationByPersonId getVisitInformationByPersonId,
+            IGetAllCaseNotesUseCase getAllCaseNotesUseCase)
+        {
+            _getVisitInformationByPersonId = getVisitInformationByPersonId;
+            _getAllCaseNotesUseCase = getAllCaseNotesUseCase;
+        }
+
+        public List<ResidentHistoricRecord> Execute(long personId)
+        {
+            var visits = _getVisitInformationByPersonId.Execute(personId);
+            var caseNotes = _getAllCaseNotesUseCase.Execute(personId).CaseNotes;
+
+            var visitRecords = (from visit in visits select visit.ToSharedResponse(personId)).ToList();
+            var caseNoteRecords = (from caseNote in caseNotes select caseNote.ToSharedResponse(personId)).ToList();
+
+            var residentHistoricRecords = new List<ResidentHistoricRecord>();
+
+            residentHistoricRecords.AddRange(visitRecords.Cast<ResidentHistoricRecord>());
```

Comment:
First time coming across .Cast method. From what I can see it's mapping the properties in the visitRecords or caseNoteRecords onto ResidentHistoricRecords, in a similar way to how the response factories do, is that correct?

Reply:
Think it is a bit different, ResidentHistoricRecords is the parent class of both ResidentHistoricRecordCaseNote and ResidentHistoricRecordVisit.

So it is less of a mapping but allowing the CaseNotes and Visits to be returned together in a single list by using a shared parent class.

It's not too different from the factory but we can do more with the factories I think.